### PR TITLE
Add GRANT_SUDO to enable sudo for job submission

### PR DESCRIPTION
**Pull request by ctiml**
Signed-off-by: Timothy Lee <ctiml@infuseai.io>
Co-authored-by: Ash Wu <hsatac@infuseai.io>

** PR checklist **

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**
Feature

**What this PR does / why we need it**:
Enable sudo for job submission

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

File: ee/scripts/job/run-job.sh [hunk 1]
```diff
@@ -69,8 +70,22 @@ copy_artifacts() {
...
+  fi
+  if [[ -n $USER ]] && [[ -d /etc/sudoers.d ]]; then
+    echo "$USER ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/job
+  fi
+fi
+
 # Run Command
-bash -c "$COMMAND"
+if [[ "$GRANT_SUDO" == "true" ]] && [[ -n $USER ]]; then
+  exec sudo -E -H -u $USER PATH=$PATH bash -c "$COMMAND"
```
Comment (popcornylu) [inline ee/scripts/job/run-job.sh:85]: Should we check if sudo command is available?.

```
if command -v sudo > /dev/null && [[ "$GRANT_SUDO" == "true" ]] && [[ -n $USER ]]; then
```

**Review by popcornylu — COMMENTED** [review]
Should check if binary `sudo` exists.  The job should not be docker-stacks image. It could be NGC, repo2docker or any source of image, we should reduce the requirement as few as possible.

File: ee/scripts/job/run-job.sh (hunk 1 above)
Comment (ctiml) [reply ee/scripts/job/run-job.sh:85]: Sounds great

**Comment by popcornylu** [comment]
LGTM

**Pull request merged**